### PR TITLE
config: keep compatibility after remove tracer

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
in #1116 we remove tracer's code, but that breaks compatibility when upgrade from old version.

### What is changed and how it works?
add a field to decode `tracer`

### Check List 

Tests 

 - Manual test
   - start a simple task in 2.0.0-rc DM
   - kill dm-master and dm-worker
   - start binary built from current master, works fine

Code changes

 - Has persistent data change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: dm/config/source_config.go [hunk 1]
```diff
@@ -71,6 +71,9 @@ type SourceConfig struct {
 
 	// id of the worker on which this task run
 	ServerID uint32 `yaml:"server-id" toml:"server-id" json:"server-id"`
+
+	// deprecated tracer, to keep compatibility with older version
+	Tracer map[string]interface{} `yaml:"tracer" toml:"tracer" json:"tracer"`
```
Comment (csuzhangxc) [inline dm/config/source_config.go:76]: How about set flag `json` to `-`?

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

**Review by csuzhangxc — DISMISSED** [review]
LGTM

**Comment by lance6716** [comment]
ci failed, will fix soon

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**